tensorflow_addons/layers/crf.py

```diff
@@ -0,0 +1,334 @@
+
+        # global variable
+        self.kernel = None
+        self.chain_kernel = None
+        self.bias = None
+        self.left_boundary = None
+        self.right_boundary = None
+
+    def build(self, input_shape):
+        input_shape = tuple(tf.TensorShape(input_shape).as_list())
+        self.input_spec = [InputSpec(shape=input_shape)]
+        self.input_dim = input_shape[-1]
+
+        # weights that mapping arbitrary tensor to correct shape
+        self.kernel = self.add_weight(shape=(self.input_dim, self.units),
+                                      name='kernel',
+                                      initializer=self.kernel_initializer,
+                                      regularizer=self.kernel_regularizer,
+                                      constraint=self.kernel_constraint)
+
+        # weights that work as transfer probability of each tags
+        self.chain_kernel = self.add_weight(shape=(self.units, self.units),
+                                            name='chain_kernel',
+                                            initializer=self.chain_initializer,
+                                            regularizer=self.chain_regularizer,
+                                            constraint=self.chain_constraint)
+
+        # bias that works with self.kernel
+        if self.use_bias:
+            self.bias = self.add_weight(shape=(self.units,),
+                                        name='bias',
+                                        initializer=self.bias_initializer,
+                                        regularizer=self.bias_regularizer,
+                                        constraint=self.bias_constraint)
+        else:
+            self.bias = 0
+
+        # weight of <START> to tag probability and tag to <END> probability
+        if self.use_boundary:
+            self.left_boundary = self.add_weight(shape=(self.units,),
+                                                 name='left_boundary',
+                                                 initializer=self.boundary_initializer,
+                                                 regularizer=self.boundary_regularizer,
+                                                 constraint=self.boundary_constraint)
+            self.right_boundary = self.add_weight(shape=(self.units,),
+                                                  name='right_boundary',
+                                                  initializer=self.boundary_initializer,
+                                                  regularizer=self.boundary_regularizer,
+                                                  constraint=self.boundary_constraint)
+
+        # or directly call self.built = True
+        super(CRF, self).build(input_shape)
+
+    def call(self, inputs, mask=None, **kwargs):
+        # mask: Tensor(shape=(?, ?), dtype=bool) or None
+
+        if mask is not None:
+            assert K.ndim(mask) == 2, 'Input mask to CRF must have dim 2 if not None'
```

Comment:
Check using `if` and raise an Error.

Reply:
What do you mean? Using if in here may raise an error?

Reply:
I think he means you should raise an Error in else case, i.e; When `mask is None`

Reply:
I think He means you should raise a ValueError (or another appropriate Error, like maybe AttributeError) when `mask is None`, i.e; 
```py
if mask is not None:
    assert K.ndim(mask) == 2, 'Input mask to CRF must have dim 2 if not None'
else:
    raise ValueError(A tensor for variable %s of type %s is required'
                         ' for layer %s' % (name, dtype.base_dtype, self.name))
```
name = mask
dtype.base_dtype = type(mask)
self.name = Arbitrary Name of the layer object (if any)

Something along theses lines, I suppose